tests/test_internals/test_roi_extractors_module.py

```diff
@@ -481,6 +482,140 @@ def test_add_two_plane_segmentation(self):
         assert second_plane_segmentation.description == second_plane_segmentation_description
 
 
+class TestAddFluorescence(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.num_rois = 10
+        cls.num_frames = 20
+        cls.num_rows = 25
+        cls.num_columns = 20
+
+        cls.session_start_time = datetime.now().astimezone()
+
+        cls.fluorescence_name = "fluorescence"
+
+        cls.raw_roi_response_series_metadata = dict(
+            name="raw",
+            description="raw fluorescence signal",
+        )
+
+        cls.dff_roi_response_series_metadata = dict(
+            name="dff",
+            description="relative (df/f) fluoresence signal",
+        )
+
+        cls.deconvolved_roi_response_series_metadata = dict(
+            name="deconvolved",
+            description="deconvolved fluorescence signal",
+        )
+
+        cls.neuropil_roi_response_series_metadata = dict(
+            name="neuropil",
+            description="neuropil fluorescence signal",
+        )
+
+    def setUp(self):
+        self.segmentation_extractor = generate_dummy_segmentation_extractor(
+            num_rois=self.num_rois,
+            num_frames=self.num_frames,
+            num_rows=self.num_rows,
+            num_columns=self.num_columns,
+        )
+        self.nwbfile = NWBFile(
+            session_description="session_description",
+            identifier="file_id",
+            session_start_time=self.session_start_time,
+        )
+
+        self.metadata = dict(Ophys=dict())
+
+        fluorescence_metadata = dict(
+            Fluorescence=dict(
+                name=self.fluorescence_name,
+                roi_response_series=[
+                    self.raw_roi_response_series_metadata,
+                    self.dff_roi_response_series_metadata,
+                    self.deconvolved_roi_response_series_metadata,
+                    self.neuropil_roi_response_series_metadata,
+                ],
+            )
+        )
+
+        self.metadata["Ophys"].update(fluorescence_metadata)
+
+    def test_add_fluorescence(self):
+        """Test adding fluorescence to the nwbfile."""
+
+        add_fluorescence(
+            segmentation_extractor=self.segmentation_extractor,
+            nwbfile=self.nwbfile,
+            metadata=self.metadata,
+        )
+
+        fluorescence = self.nwbfile.processing["ophys"].get(self.fluorescence_name)
+
+        self.assertEqual(fluorescence.name, self.fluorescence_name)
+        self.assertEqual(len(fluorescence.roi_response_series), len(self.segmentation_extractor.get_traces_dict()))
+
+    def test_add_fluorescence_one_of_the_traces_is_none(self):
```

Comment:
The `dummy_segmentation_extractor` has functionality for building examples where the traces are not available so you don't need to set the private attribute to None here:

https://github.com/catalystneuro/roiextractors/blob/1f8a479e90a45595de661ad339fa53fc022294e2/src/roiextractors/testing.py#L54-L66

Reply:
ah thank you!